### ISSUE TITLE
panicked at 'assertion failed: value.is_object() || value.is_function()'

### ISSUE THREAD

**Issue by charlesxsh** [boa-dev/boa#1404]
**Describe the bug**
thread 'main' panicked at 'assertion failed: value.is_object() || value.is_function()'

**To Reproduce**
Input: 
`heD=eD.hD=D=uD=heD=NaN=eD=hvh=h=heheD=ODeee=h=hAuheAle1=heA=e1heAhhehheA=heA=e1heAhueAhe=44444444`

Code:
```rust
fn main(){
    // let data = input

    let mut context = Context::new();
    context.eval(data);

}
```

Output
```
thread 'main' panicked at 'assertion failed: value.is_object() || value.is_function()', /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/environment/object_environment_record.rs:77:9
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::begin_panic
             at /home/xsh/code/rust-compiler/src/libstd/panicking.rs:456
   7: <boa::environment::object_environment_record::ObjectEnvironmentRecord as boa::environment::environment_record_trait::EnvironmentRecordTrait>::set_mutable_binding
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/environment/object_environment_record.rs:77
   8: <boa::environment::global_environment_record::GlobalEnvironmentRecord as boa::environment::environment_record_trait::EnvironmentRecordTrait>::set_mutable_binding
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/environment/global_environment_record.rs:186
   9: boa::environment::lexical_environment::LexicalEnvironment::set_mutable_binding
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/environment/lexical_environment.rs:200
  10: <boa::syntax::ast::node::operator::assign::Assign as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/operator/assign/mod.rs:65
  11: <boa::syntax::ast::node::Node as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/mod.rs:325
  12: <boa::syntax::ast::node::operator::assign::Assign as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/operator/assign/mod.rs:58
  13: <boa::syntax::ast::node::Node as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/mod.rs:325
  14: <boa::syntax::ast::node::operator::assign::Assign as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/operator/assign/mod.rs:58
  15: <boa::syntax::ast::node::Node as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/mod.rs:325
  16: <boa::syntax::ast::node::operator::assign::Assign as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/operator/assign/mod.rs:58
  17: <boa::syntax::ast::node::Node as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/mod.rs:325
  18: <boa::syntax::ast::node::operator::assign::Assign as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/operator/assign/mod.rs:58
  19: <boa::syntax::ast::node::Node as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/mod.rs:325
  20: <boa::syntax::ast::node::operator::assign::Assign as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/operator/assign/mod.rs:58
  21: <boa::syntax::ast::node::Node as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/mod.rs:325
  22: <boa::syntax::ast::node::statement_list::StatementList as boa::exec::Executable>::run
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/syntax/ast/node/statement_list/mod.rs:71
  23: boa::context::Context::eval
             at /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/Boa-0.11.0/src/context.rs:717

```

**Expected behavior**
eval should return error properly instead of panic

**Build environment (please complete the following information):**

- OS: Ubuntu
- Version: 18.04
- Target triple: x86_64-unknown-linux-gnu
- Rustc version: rustc 1.46.0

**Additional context**

**Comment by neeldug** [comment]
Related to #948

**Comment by HalidOdat** [comment]
Fixed by #1569
